cmsl1t/playground/lumifilters.py

```diff
@@ -0,0 +1,22 @@
+import urllib2
+import json
+
+def lumiFilter(run, lumi, lumiJson=''):
+
+    # If no lumi mask json, pass all events    
+    if not lumiJson:
+        return True
+
+    # Load json
+    if 'https://' in lumiJson:
+        response = urllib2.urlopen(lumiJson)
+        lumiMask = json.load(response)
+    else:
+        lumiMask = json.load(open(lumiJson))
+
+    if str(run) in lumiMask:
+        for ls_list in lumiMask[str(run)]:
+            if lumi >= ls_list[0] and lumi <= ls_list[1]:
```

Comment:
That is incorrect, is it not? Removed lumis will be still accepted here

Reply:
@benkrikler I am trying to think of a good method for fast lookups. A sparse matrix would be nice, but ideally we would keep the dimensions low:
 1. initialise matrix with zeros
 2. store min & max run and lumi
 3. normalise to run and lumi range
 4. fill ones where we have entries

Reply:
Or howabout a python set where the keys are just a tuple of the run number and lumi section?  We fill the set initially by expanding the lower and upper lumi section limits for each run in the same as you'd do for the matrix above.  And since the `(run, lumisection)` tuple will just be hashed and python's internal find mechanism being so fast, I think that should perform just as well or slightly better even than a matrix, but be fairly easy to code:

```python
def __init__(self, run_lumis_dict):
   self.valid_lumi_sections = set()
   for run, lumi_ranges in run_lumis_dict:
      for lumi_first, lumi_last in lumi_ranges:
          for lumi in range(lumi_first, lumi_last):
              self.valid_lumi_sections.insert( (run, lumi) )

def is_valid_run_lumi(self, run, lumi):
    return (run, lumi) in self.valid_lumi_sections
```

Reply:
That sounds like a reasonable first step. I would add a benchmark for this.